src/components/header/index.tsx

```diff
@@ -1,25 +1,37 @@
 import { ReactComponent as ProfileIcon } from '../../assets/profile.svg';
+import { ROUTE_PATH } from '../../routes';
 import { ReactComponent as RingIcon } from '../../assets/ring.svg';
 import { ReactComponent as SearchIcon } from '../../assets/search.svg';
 import { links } from '../navBar/helper';
-import { useLocation } from 'react-router-dom';
 import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { user } from '../../store/user/selectors';
+import { Dropdown, Menu, MenuItem } from '@mui/base';
 import {
   HeaderContainer,
   IconsContainer,
-  PhotoContainer,
+  Listbox,
   ProfileContainer,
+  StyledButton,
   StyledHeaderText,
   StyledHr,
+  StyledMenuButton,
   StyledTypography,
 } from './styled';
+import { useLocation, useNavigate } from 'react-router-dom';
 
 const Header: React.FC = () => {
+  const currentUser = useSelector(user);
   const location = useLocation();
+  const navigate = useNavigate();
   const pageName = useMemo(
     () => links.find((link) => link.to === location.pathname)?.name ?? '',
     [location.pathname],
   );
+  const createHandleMenuClick = () => {
```

Comment:
let's create the separate action for logout. Which will clean up localStorage state etc.

Reply:
fix: localStorage  and state operations moved to action